src/Autofac/Core/Resolving/Pipeline/IResolveRequestContext.cs

```diff
@@ -0,0 +1,140 @@
+﻿// This software is part of the Autofac IoC container
+// Copyright © 2020 Autofac Contributors
+// https://autofac.org
+//
+// Permission is hereby granted, free of charge, to any person
+// obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Autofac.Core.Diagnostics;
+using Autofac.Core.Registration;
+
+namespace Autofac.Core.Resolving.Pipeline
+{
+    /// <summary>
+    /// Defines the context object for a single resolve request. Provides access to the in-flight status of the operation,
+    /// and ways to manipulate the contents.
+    /// </summary>
+    public interface IResolveRequestContext : IComponentContext
```

Comment:
I suppose it's similar to how the ASP.NET Core context type is `HttpContextBase`? I guess we can be consistent with the operation here.

Reply:
I think so, and it could probably make it easier to enhance later. I know there are default interface implementations, but... it kinda gives me the heebie-jeebies.

Reply:
Ok; I'll define a `ResolveRequestContextBase` and pass that.

Reply:
`IResolveRequestContext` -> `ResolveRequestContextBase`.  `ResolveComponent` and `ResolveComponentWithNewOperation` are now abstract methods implemented by `ResolveRequestContext`.